core/core-impl/src/test/java/ai/timefold/solver/core/api/solver/SolverManagerTest.java

```diff
@@ -213,6 +217,59 @@ void solveGenerics() throws ExecutionException, InterruptedException {
         solverJob.getFinalBestSolution();
     }
 
+    @Test
+    void solveWithOverride() throws InterruptedException {
+        CountDownLatch countDownLatch = new CountDownLatch(2);
+        PhaseConfig<?> customPhaseConfig = new CustomPhaseConfig().withCustomPhaseCommands(
+                scoreDirector -> {
+                    await().pollDelay(Duration.ofSeconds(5L)).until(() -> true);
+                    countDownLatch.countDown();
```

Comment:
We typically try to avoid unit tests which run for seconds. Unit tests are supposed to be cheap and quick, otherwise people stop running them. Arguably you can have the wait much shorter, and then in the end check for how long the solver actually ran.

(Note: every time you play with time in tests, you inevitably invite the test to be flaky.)

Reply:
Yes, it makes sense. The goal is to ensure the termination time is overridden, but the internal state is not fetchable. I'll try to think of something else.

Reply:
I simplified the test but had to expose `Termination<Solution_>` on `DefaultSolverJob`.